src/main/java/hudson/plugins/gradle/injection/MavenInjectionAware.java

```diff
@@ -0,0 +1,96 @@
+
+    // Maven system properties passed on the CLI to a Maven build
+    SystemProperty.Key GRADLE_ENTERPRISE_URL_PROPERTY_KEY = SystemProperty.Key.required("gradle.enterprise.url");
+    SystemProperty.Key BUILD_SCAN_UPLOAD_IN_BACKGROUND_PROPERTY_KEY = SystemProperty.Key.required("gradle.scan.uploadInBackground");
+    SystemProperty.Key MAVEN_EXT_CLASS_PATH_PROPERTY_KEY = SystemProperty.Key.required("maven.ext.class.path");
+    SystemProperty.Key GRADLE_ENTERPRISE_ALLOW_UNTRUSTED_SERVER_PROPERTY_KEY = SystemProperty.Key.optional("gradle.enterprise.allowUntrustedServer");
+
+    MavenOptsHandler MAVEN_OPTS_HANDLER = new MavenOptsHandler(
+            MAVEN_EXT_CLASS_PATH_PROPERTY_KEY,
+            BUILD_SCAN_UPLOAD_IN_BACKGROUND_PROPERTY_KEY,
+            GRADLE_ENTERPRISE_ALLOW_UNTRUSTED_SERVER_PROPERTY_KEY,
+            GRADLE_ENTERPRISE_URL_PROPERTY_KEY
+    );
+
+    default boolean isInjectionDisabledGlobally(InjectionConfig config) {
+        if (config.isDisabled() || !config.isInjectMavenExtension()) {
+            return true;
+        }
+
+        return InjectionUtil.isInvalid(InjectionConfig.checkRequiredUrl(config.getServer()));
+    }
+
+    default boolean isInjectionEnabledForNode(Node node) {
+        InjectionConfig config = InjectionConfig.get();
+
+        if (isInjectionDisabledGlobally(config)) {
+            return false;
+        }
+
+        Set<String> disabledNodes =
+                CollectionUtil.safeStream(config.getMavenInjectionDisabledNodes())
+                        .map(NodeLabelItem::getLabel)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toSet());
+
+        Set<String> enabledNodes =
+                CollectionUtil.safeStream(config.getMavenInjectionEnabledNodes())
+                        .map(NodeLabelItem::getLabel)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toSet());
+
+        return InjectionUtil.isInjectionEnabledForNode(node::getAssignedLabels, disabledNodes, enabledNodes);
+    }
+
+    default void setPreparedMavenProperties(Run<?, ?> run, TaskListener listener) {
+        try {
+            EnvVars environment = run.getEnvironment(listener);
+
+            String preparedMavenOpts = environment.get(JENKINSGRADLEPLUGIN_MAVEN_OPTS_PREPARED);
+            String preparedMavenPluginConfigExtClasspath = environment.get(JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_EXT_CLASSPATH_PREPARED);
+            if (preparedMavenOpts != null && preparedMavenPluginConfigExtClasspath != null) {
+                run.addAction(new NodePreparedMavenEnvsAction(preparedMavenOpts, preparedMavenPluginConfigExtClasspath));
+            }
+        } catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Action that holds Maven environment variables to be set in EnvironmentContributor
```

Comment:
I think we need to reference a concrete contributor in the javadoc, not the abstract contributor, as this is a very specific action:

```suggestion
     * Action that holds Maven environment variables to be set in {@link MavenInjectionEnvironmentContributor}.
```

Reply:
Done